include/deal.II/matrix_free/fe_evaluation_base_data.h

```diff
@@ -371,42 +404,85 @@ class FEEvaluationBaseData
   unsigned int
   get_active_quadrature_index() const;
 
-  //@}
+  /**
+   * If is_face is true, this function returns the face number within a cell for
+   * the face this object was initialized to. On cells where the face makes no
+   * sense, an exception is thrown.
+   *
+   * @note This function depends on the internal representation of data, which
+   * is not stable between releases of deal.II, and is hence mostly for
+   * internal use.
+   */
+  unsigned int
+  get_face_no() const;
```

Comment:
I am wondering why this is `unsigned int` and not `std::array<unsigned int, X>`? How else do we want to support ECL without to have to write specialized code at many places?